Automattic/wp-calypso
# WPCOM: Migrate `wpcom.undocumented()` site plans to `wpcom.req`

#### Changes proposed in this Pull Request

This PR migrates all the `wpcom.undocumented()` site plans method to `wpcom.req.get()`. 

The PR also simplifies the code a little bit - we don't need to encode the site fragment, because anywhere in the code where we're making requests to this endpoint, we pass the numeric site ID. There is no longer a reason to support non-numeric IDs there - we removed that support a while ago. We're also removing an unnecessary wrapping `Promise` - `wpcom.req.get()` already returns a promise. 

We also use the opportunity to remove the unused `cancelSitePlanTrial` action creator, its related reducer and tests and the unused undocumented `cancelPlanTrial` method. It was created almost 6 years ago in #2952 and it hasn't been used yet, so it's time to go.

Part of the ongoing effort to get rid of `wpcom.undocumented()`.

#### Testing instructions

* Go to `/plans/:site` where `:site` is one of your Jetpack sites.
* Verify that the request to `/sites/:site/plans` is still successful.
* Verify the removed code is not in use.
